substrate/client/basic-authorship/src/basic_authorship.rs

```diff
@@ -291,7 +291,7 @@ where
 			None,
 			Box::pin(async move {
 				// leave some time for evaluation and block finalization (33%)
-				let deadline = (self.now)() + max_duration - max_duration / 3;
+				let deadline = (self.now)() + max_duration - max_duration / 5;
```

Comment:
But we cannot merge this change. Or can we? @bkchr? @sandreim?

Reply:
I'd lean towards no. I don't even know how we got that 33%  number. AFAIK in block finalization we recompute the state root and this shouldn't take that much (600ms), @bkchr can you provide more insight ?

Reply:
Block finalization is clearly more than just computing the `storage_root`. We are running `on_idle`, doing clean ups or other stuff. If the machine is following the recommended hardware, it should be fine. Because the weight for `on_finalize` + is accounted for in `on_initialize` and `on_idle` just takes what is left. So, if benchmarking is working correctly, everything should be fine. But yeah right now it also accounts for other stuff like compressing etc.

TLDR: Decreasing that should be fine, maybe even put it down to `200ms` for `2s` block time.

Reply:
I suggest we add a CLI argument for the percentage here defaulting to to 33%

Reply:
Actually, not needed to happen in this PR, will create a separate ticket.